vignettes/irecist.Rmd

```diff
@@ -0,0 +1,877 @@
+function you can optionally decide what period would you consider an `iSD` or
+`NON-iCR/NON-iUPD` as being eligible from. In this example, 42 days after
+randomization date has been used again.
+
+Please note that the order of the events specified for `events` is important.
+For example, a subject with `iPR`, `iPR`, `iCR` qualifies for both `ibor_icr` and
+`ibor_ipr`. As `ibor_icr` is listed before `ibor_ipr`, `iCR` is selected as best overall
+response for this subject.
+
+```{r}
+ibor_icr <- event(
+  description = "Define complete response (iCR) for best overall response (iBOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "iCR",
+  set_values_to = exprs(AVALC = "iCR")
+)
+
+ibor_ipr <- event(
+  description = "Define partial response (iPR) for best overall response (iBOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "iPR",
+  set_values_to = exprs(AVALC = "iPR")
+)
+
+ibor_isd <- event(
+  description = paste(
+    "Define stable disease (iSD) for best overall response (iBOR) as iCR, iPR, or iSD",
+    "occurring at least 42 days after randomization"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC %in% c("iCR", "iPR", "iSD") & ADT >= RANDDT + 42,
+  set_values_to = exprs(AVALC = "iSD")
+)
+
+ibor_non_icriupd <- event(
+  description = paste(
+    "Define NON-iCR/NON-iUPD for best overall response (iBOR) as NON-iCR/NON-iUPD",
+    "occuring at least 42 days after randomization"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC == "NON-iCR/NON-iUPD" & ADT >= RANDDT + 42,
+  set_values_to = exprs(AVALC = "NON-iCR/NON-iUPD")
+)
+
+
+ibor_icpd <- event_joined(
+  description = paste(
+    "Define confirmed progressive disease (iCPD) for best overall response (iBOR) as",
+    "iUPD followed by iCPD with only other iUPD and NE responses in between"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  first_cond_upper = AVALC.join == "iCPD",
+  condition = AVALC == "iUPD" &
+    all(AVALC.join %in% c("iCPD", "iUPD", "NE")),
+  set_values_to = exprs(AVALC = "iCPD")
+)
+
+
+ibor_iupd <- event(
+  description = "Define unconfirmed progressive disease (iUPD) for best overall response (iBOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "iUPD",
+  set_values_to = exprs(AVALC = "iUPD")
+)
+
+ibor_ne <- event(
+  description = paste(
+    "Define not evaluable (NE) for best overall response (iBOR) as iSD, NON-iCR/NON-iUPD,",
+    "or NE (should be specified after ibor_isd and ibor_non_icriupd)"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC %in% c("iSD", "NON-iCR/NON-iUPD", "NE"),
+  set_values_to = exprs(AVALC = "NE")
+)
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    tmp_event_nr_var = event_nr,
+    order = exprs(event_nr,ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(ibor_icr, ibor_ipr, ibor_isd, ibor_non_icriupd, ibor_iupd, ibor_icpd, ibor_ne,  no_data_missing),
```

Comment:
ibor_icpd should be before ibor_iupd, so in case no better responses then iUPD and iCPD we would assing iCPD as iBOR

Reply:
updated